### PR TITLE
WIP: Drugs changes

### PR DESCRIPTION
**Pull request by CthulhuOnIce**
Adds blotter, which can be created by using a glass bottle or beaker on a paper, 
and acid, which doesn't have a recipe (yet) and fucks with your colors, and hopefully more stuff soon

### CODE REVIEW

File: code/modules/reagents/chemistry/reagents/drug_reagents.dm [hunk 1]
```diff
@@ -439,3 +439,42 @@
...
 	. = 1
+
+/datum/reagent/drug/acid
+	name = "Acid"
+	description = "A strong psychedelic drug."
+	reagent_state = LIQUID
+	color = "#B0B0B0"
+	taste_description = "bitterness"
+
+/datum/reagent/drug/acid/on_mob_life(mob/living/M)
```
Comment (qwertyquerty) [inline code/modules/reagents/chemistry/reagents/drug_reagents.dm]: Technically you could use a mixture of spawn while sleep to get it to update more often than every 2 seconds

File: code/modules/reagents/reagent_containers/blotter.dm [hunk 2]
```diff
@@ -0,0 +1,5 @@
+/obj/item/reagent_containers/pill/blotter
+	name = "Blotter"
+	desc = "Paper soaked in a reagent you can take via mouth."
+	icon_state = "blotter"
+	volume = 50
```
Comment (qwertyquerty) [inline code/modules/reagents/reagent_containers/blotter.dm]: Big volume for just a little piece of paper

File: code/modules/reagents/chemistry/reagents/drug_reagents.dm [hunk 3]
```diff
@@ -439,3 +439,42 @@
...
+			final = color_matrix_contrast(contrast)
+
+
+		M.client.color = final
+
+	return ..()
+
+/datum/reagent/drug/acid/on_mob_delete(mob/living/M)
+	if(M.client)
+		M.client.color = null
```
Comment (qwertyquerty) [inline code/modules/reagents/chemistry/reagents/drug_reagents.dm]: `M?.client?.color = null`

File: code/modules/reagents/chemistry/reagents/drug_reagents.dm (hunk 1 above)
Comment (CthulhuOnIce) [reply code/modules/reagents/chemistry/reagents/drug_reagents.dm]: that'd be hacky

**Comment by CthulhuOnIce** [comment]
Blotter: https://i.imgur.com/990KCcU.gifv

**Comment by CthulhuOnIce** [comment]
Acid: https://i.imgur.com/N8Hzcq7.gifv

**Comment by CthulhuOnIce** [comment]
LSD also does hue shifts to change the color of objects, those yellow / pink / blue tiles in that video are actually red

File: code/modules/reagents/chemistry/reagents/drug_reagents.dm [hunk 4]
```diff
@@ -439,3 +439,42 @@
...
+	
+			degree = rand(60, ((36/1000)*volume))
+			if(prob(50))	degree = -degree
+			var/list/hue_rotation = color_matrix_rotate_hue(degree)
+
+			final = color_matrix_add(hue_rotation, color_matrix_add(contrast, saturation))
+
+		else
+			var/div = rand(10,20)
+			var/contrast = volume/div < 1.1 ? volume/div : 1.1 // dont let things gray out, it always needs to be slightly contrasted
```
Comment (qwertyquerty) [inline code/modules/reagents/chemistry/reagents/drug_reagents.dm]: ouch.

`min(volume/div, 1.1)`

File: code/modules/reagents/chemistry/reagents/drug_reagents.dm (hunk 3 above)
Comment (CthulhuOnIce) [reply code/modules/reagents/chemistry/reagents/drug_reagents.dm]: Only one of those ?.'s is necessary, since client always has the color var

Comment (qwertyquerty) [reply code/modules/reagents/chemistry/reagents/drug_reagents.dm]: `M?.client.color = null`

**Comment by CthulhuOnIce** [comment]
TODO:

- [ ] Chemistry recipes for the drugs
- [ ] Audio hallucinations
- [ ] More drugs
- [ ] Powder form maybe?
- [x] Different blotter designs and patterns
- [x] Blotter renaming

File: code/modules/reagents/chemistry/reagents/drug_reagents.dm (hunk 3 above)
Comment (CthulhuOnIce) [reply code/modules/reagents/chemistry/reagents/drug_reagents.dm]: yeah this is pointless

File: code/modules/reagents/reagent_containers/blotter.dm (hunk 2 above)
Comment (CthulhuOnIce) [reply code/modules/reagents/reagent_containers/blotter.dm]: lots of drugs to take, plus you cant remove the reagent from the blotter without diluting it

**Comment by qwertyquerty** [comment]
You don't have to make a new commit for every little change you make

**Comment by CthulhuOnIce** [comment]
how've i done so far dad

**Comment by qwertyquerty** [comment]
you do not need 15 commits for the amount of code you've written

**Comment by CthulhuOnIce** [comment]
it's WIP, as in this is a rolling PR with tweaks and changes being made often, sorry if that offends you

**Comment by qwertyquerty** [comment]
It greatly offends me

**Comment by CthulhuOnIce** [comment]
@qwertyquerty this is fine to merge with the stuff i already have added

**Comment by CthulhuOnIce** [comment]
wait where did these conflicts come from

**Comment by CthulhuOnIce** [comment]
tbh

**Pull request closed without merging**